### ISSUE TITLE
Can not Login and stuck at "Complete 2-factor authentication"

### ISSUE THREAD

**Issue by milannakum** [MicrosoftDocs/vscodespaces#174]
Can't move further from login screen. Attaching the image for more clarification on the actual issue.

![Screenshot (12)](https://user-images.githubusercontent.com/11374368/68754689-30c91180-062d-11ea-9f6e-286cf0ba1a84.png)

**Comment by ljtill** [comment]
+1 : Also encountering this issue (Microsoft account).

**Comment by legomushroom** [comment]
Can you describe what happens for you? When you click the `Complete 2-factor authorization`, does the popup show up? If you go through the step, it redirects you back?

Having precise repro steps will help a lot. Thanks!

**Comment by milannakum** [comment]
> Can you describe what happens for you? When you click the `Complete 2-factor authorization`, does the popup show up? If you go through the step, it redirects you back?
> 
> Having precise repro steps will help a lot. Thanks!

Exactly. After clicking Complete 2-factor, it shows popup to login, I login, then it redirects back to the same page.

**Comment by TethysSvensson** [comment]
I am having the same issue (or at least one with the same symptoms). I do not have 2-factor authentication enabled as far as I know.

I managed to get past this issue by temporarily disabling the setting "Block third-party cookies" in my browser.

**Comment by milannakum** [comment]
I am able to login today without getting stuck at Complete 2-factor Auth.

**Comment by TethysSvensson** [comment]
I am still not able to log in without enabling 3rd party cookies.

**Comment by legomushroom** [comment]
@TethysSvensson we have a change in a review that should eliminate the need for the 3rd party cookies on VSO side for authentication. I'll reopen this issue to keep track of this. That said, AAD relies on cookies for auth so you might have issues with AAD authentication flows, let me perform few tests.

**Comment by zchwei** [comment]
@legomushroom +1 : Also encountering this issue when attempting **second** to login (Microsoft account) after delete vsonline resource in Azure.

**Comment by zeekrey** [comment]
Same issue here. Since my company admins disabled 3rd party cookies I'm not able to load VSCode Online. :(

**Comment by jmbockhorst** [comment]
This is also happening to me when logging in on Safari on iPadOS.

**Comment by jsejcksn** [comment]
This is a third-party cookies dependency issue. Please make any necessary cookies first-party.

![vsonline-issue-3rd_party_cookies](https://user-images.githubusercontent.com/868251/74556986-caea6f80-4f24-11ea-8a86-a11d645b4613.gif)

**Comment by owingruters** [comment]
+1

**Comment by mistryrn** [comment]
Also running into this issue on Safari on iPadOS. Tried disabling "Prevent Cross-Site Tracking" in the Safari settings like @jmbockhorst suggested and was eventually able to sign in.

**Comment by lostintangent** [comment]
@legomushroom Any updates here?

**Comment by legomushroom** [comment]
This should be fixed by now with @klvnraju changes to the auth flow. 3rd party cookies are still an issue though.

**Comment by lostintangent** [comment]
@legomushroom Don't we have a separate issue tracking the 3rd-party cookies? If so, can we resolve this and reference the cookie issue?

**Comment by jsejcksn** [comment]
> can we resolve this and reference the cookie issue?

@lostintangent You closed this issue, but I don't see a link to the issue about third-party cookies. Will you post it here?

**Comment by klvnraju** [comment]
This is another issue for 3rd party cookies.
https://github.com/MicrosoftDocs/vscodespaces/issues/345

**Comment by lostintangent** [comment]
@jsejcksn That was totally my mistake, since I didn't intend to close this issue 😄 @legomushroom Feel free to close this issue if you think that #345 is capturing the remaining work already. Thanks!

**Comment by legomushroom** [comment]
I think this issue is more accurate for tracking the 3rd party cookies so I'd keep it open so other folks can find the solution.
I don't think we will invest in fixing this issue though since that's a big change and the VSO service gets sunset in favor of GitHub Codespaces which will render this issue obsolete.
